### ISSUE TITLE
No exported member 'TextDocumentWillSaveEvent'

### ISSUE THREAD

**Issue by Turbo87** [Microsoft/vscode-languageserver-node#151]
I'm seeing the following error when compiling my client project with `vscode-languageclient@3.0.3`:

```
node_modules/vscode-languageclient/lib/codeConverter.d.ts(12,46): error TS2694: Namespace 'vscode' has no exported member 'TextDocumentWillSaveEvent'.
```

**Comment by Turbo87** [comment]
It seems like this is related to our `engines` setting in `package.json` specifying `"vscode": "^1.4.0"`, which apparently doesn't have that event type yet.

**Comment by dbaeumer** [comment]
@Turbo87 I am a little puzzled. The 3.0.3 version of the client has 
```json
	"engines": {
		"vscode": "^1.8.0"
	}
```

**Comment by Turbo87** [comment]
@dbaeumer I have to admit I only understand half of what I'm experimenting with at the moment... 😄 

tl;dr is that https://github.com/emberwatch/ember-language-server/pull/13/commits/8632132d1673b1f7db168c89420a2be7919a985b solved our problem after we upgraded our client to `^3.0.0`

**Comment by dbaeumer** [comment]
Now I see. NPM doesn't do any 'recursive' engine checking nor do we. So if the languageclient needs 1.8.0 you need to reiterate this in your extension as well. 

We were debating whether the language client should throw on an engine mismatch but we decided against it since node itself doesn't through either. If you feel strong about the client throwing, please open a corresponding issue.

**Comment by Turbo87** [comment]
> If you feel strong about the client throwing, please open a corresponding issue.

As I said I'm not that deeply familiar with how things work yet so having a strong opinion would probably be a stupid idea 😉 

It might make sense though to add that requirement to the README or changelog.

**Comment by dbaeumer** [comment]
See #153
